modules/xfem/include/userobjects/MeshCut3DUserObject.h

```diff
@@ -0,0 +1,78 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef MESH_CUT_3D_USEROBJECT_H
+#define MESH_CUT_3D_USEROBJECT_H
+
+#include "GeometricCutUserObject.h"
+
+// Forward declarations
+class MeshCut3DUserObject;
+
+template <>
+InputParameters validParams<MeshCut3DUserObject>();
+
+class MeshCut3DUserObject : public GeometricCutUserObject
+{
+public:
+  MeshCut3DUserObject(const InputParameters & parameters);
+  ~MeshCut3DUserObject();
+  virtual bool active(Real time) const;
+  virtual bool
+  cutElementByGeometry(const Elem * elem, std::vector<CutEdge> & cut_edges, Real time) const;
+  virtual bool
+  cutElementByGeometry(const Elem * elem, std::vector<CutFace> & cut_faces, Real time) const;
+  virtual bool cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_edges,
+                                     std::vector<CutEdge> & cut_edges,
+                                     Real time) const;
+  virtual bool cutFragmentByGeometry(std::vector<std::vector<Point>> & frag_faces,
+                                     std::vector<CutFace> & cut_faces,
+                                     Real time) const;
+
+protected:
+  std::unique_ptr<MeshBase> _cut_mesh;
+  std::vector<int> _boundary;
```

Comment:
We have a type for `boundaryID` defined in MOOSE. You should probably use that here instead of `int`.